uniffi/src/interface/mod.rs

```diff
@@ -533,9 +534,24 @@ pub struct Object {
     name: String,
     constructors: Vec<Constructor>,
     methods: Vec<Method>,
+    object_free: Method,
```

Comment:
Does this need to be an API-level `Method`? Naively, it feels like it would suffice to generate just the `FFIFunction` and rely on the automatically-generated foreign-language code for the API-level definition.

In practice that would look very similar to how `bytebuffer_free` and friends work on the `ComponentInterface`:

* The `Object::reserved_object_free` method creates a new `FFIFunction` every time it's called.
* The `Object::iter_ffi_function_definitions` method calls `Object::reserved_object_free` and appends the result to the vec of `FFIFunction` definitions.
* For foreign-language templates directly emit an FFI-level call to the function returned by `Object::reserved_object_free`.

I *think* that would be all you need, and would simplify some of the changes here.

Reply:
I've moved this as directed here; it did mean touching the foreign-language macro templates for all functions and methods, and adding a `method_arguments()` method to `FFIFunction` which elides the first argument.

Reply:
Is it now safe to remove this `object_free` property on the struct?